include/dxc/HLSL/HLSLExtensionsCodegenHelper.h

```diff
@@ -66,10 +67,13 @@ class HLSLExtensionsCodegenHelper {
 
   // Write semantic defines as metadata in the module.
   virtual void WriteSemanticDefines(llvm::Module *M) = 0;
-  virtual void UpdateCodeGenOptions(clang::CodeGenOptions &CGO) = 0;
+  // Reads semantic defines and translates them into opt-enable/opt-disable toggles. This only
+  // works after parser is initialized.
+  virtual void UpdateSemanticDefinesAndOptToggles() = 0;
   // Query the named option enable
   // Needed because semantic defines may have set it since options were copied
-  virtual bool IsOptionEnabled(std::string option) = 0;
+  virtual bool IsOptionEnabled(hlsl::options::Toggle toggle) = 0;
+  virtual bool IsLifetimeMarkersEnabled() = 0;
```

Comment:
Why is lifetime markers the only one that gets an explicit method (`IsLIfetimeMarkersEnabled`)? All the others use `IsOptionEnabled`.

Reply:
Because it checks both the flag (based on the -enable/disable-lifetime-markers flag and the -opt-enable/disable toggle).